packages/ui/src/routes/bridge/BridgeConfirmPage.tsx

```diff
@@ -232,24 +240,44 @@ const BridgeConfirmPage: FunctionComponent<{}> = () => {
         ? hasNativeAssetBalance
         : hasNativeAssetBalance && hasFromTokenBalance
 
-    const nativeTokensInDestinationNetworkStatus =
-        useAddressHasEnoughNativeTokensToSend(toChainId, toToken.address)
+    const { hideBridgeInsufficientNativeTokenWarning } = useUserSettings()
+    const { isLoading, result, check } =
+        useAddressHasEnoughNativeTokensToSend(toChainId)
 
+    const destinationTokenIsNative = isNativeTokenAddress(toToken.address)
     const destinationNetwork = Object.values(availableNetworks).find(
         (n) => n.chainId === toChainId
     )
-    const bridgeWarningMessage = getWarningMessages(
-        nativeTokensInDestinationNetworkStatus.result,
-        destinationNetwork
-    )
+    useEffect(() => {
+        if (
+            !hideBridgeInsufficientNativeTokenWarning &&
+            !destinationTokenIsNative
+        ) {
+            check()
+        }
+    }, [])
 
-    // it shows de warning when the user doesn't have enough native tokens in the destination network to perform a send and there is no pending transaction
-    const showDestinationFeeWarning =
-        !nativeTokensInDestinationNetworkStatus.isLoading &&
-        !inProgressAllowanceTransaction?.id &&
-        !inProgressTransaction?.id &&
-        nativeTokensInDestinationNetworkStatus.result !==
-            EnoughNativeTokensToSend.ENOUGH
+    // // it shows de warning when the user doesn't have enough native tokens in the destination network to perform a send and there is no pending transaction
+    useEffect(() => {
```

Comment:
You can transform this `useEffect` in an `useMemo` or even a variable since it is not a complex calculation what it does.

Also, I think it would be better to be a little bit more explicit what we mean:

```javascript
const busyScreen = inProgressAllowanceTransaction?.id || inProgressTransaction?.id 
const bridgeWarningMessage = hideBridgeInsufficientNativeTokenWarning || isLoadingUserHasEnough ? '' :   getWarningMessages(result, destinationNetwork)
const showBridgeWarningMessage  = !busyScreen && bridgeWarningMessage
```

```